awx/main/access.py

```diff
@@ -565,10 +565,7 @@ class OAuth2ApplicationAccess(BaseAccess):
     select_related = ('user',)
 
     def filtered_queryset(self):
-        accessible_users = User.objects.filter(
-            pk__in=self.user.admin_of_organizations.values('member_role__members')
-        ) | User.objects.filter(pk=self.user.pk)
-        return self.model.objects.filter(user__in=accessible_users)
+        return self.model.objects.filter(organization__in=self.user.organizations)
 
     def can_change(self, obj, data):
         return self.can_read(obj)
```

Comment:
I think that more fixes are needed to its access class. Wouldn't it make sense to have `can_admin` check if the user is an admin of the organization, and then have change & delete fall back to that?

`can_add` would also need some updating.